add supabase + pgvector datastore provider

egor-romanov opened the pull request:
## What kind of change does this PR introduce?

- add supabase as a datastore provider
- add necessary migrations with
  - Postgres function to search closest vectors using pgvector extension
  - table to store embeddings
  - enable pgvector postgres extension
- update readme

js-ts committed:
Add docker compose install link and fix the command

docker-compose got deprecated we should instead use docker compose

From the end of June 2023 Compose V1 won’t be supported anymore and will be removed from all Docker Desktop versions.

Make sure you switch to Compose V2 with the docker compose CLI plugin or by activating the Use Docker Compose V2 setting in Docker Desktop. For more information, see the Evolution of Compose

bojieli committed:
small typo fixes to code comments and README

yunfengsay committed:
Update date.py

remove unused print function

Commit:
fix test command in README.md

tylerhutcherson committed:
add redis module check

eltociear committed:
Fix typo in milvus_datastore.py

Enviroment -> Environment

egor-romanov committed:
add supabase + pgvector datastore provider

J0 commented on README.md:
```diff
@@ -491,6 +491,30 @@ Environment Variables:
+| `SUPABASE_KEY`   | Yes      | Supabase Project API key (anon or service)   |            |
+| `INIT_DB`        | Optional | Set to true if you want to run DB migrations | `False`    |
+| `PG_HOST`        | Optional | Postgres host                                | none       |
+| `PG_PORT`        | Optional | Postgres port                                | `6543`     |
```
/nit not sure if we want to add an explanatory note that we're using pg_bouncer to explain why we're using port 6543 instead of 5432 - suspect that we might get some questions

kiwicopple commented on README.md:
```diff
@@ -101,9 +101,9 @@ The plugin exposes the following endpoints for upserting, querying, and deleting
 - `/delete`: This endpoint allows deleting one or more documents from the vector database using their IDs, a metadata filter, or a delete_all flag. The endpoint expects at least one of the following parameters in the request body: `ids`, `filter`, or `delete_all`. The `ids` parameter should be a list of document IDs to delete; all document chunks for the document with these IDS will be deleted. T ...
 
-The detailed specifications and examples of the request and response models can be found by running the app locally and navigating to http://0.0.0.0:8000/openapi.json, or in the OpenAPI schema [here](/.well-known/openapi.yaml). Note that the OpenAPI schema only contains the `/query` endpoint, because that is the only function that ChatGPT needs to access. This way, ChatGPT can use the plugin only ...
+The detailed specifications and examples of the request and response models can be found by running the app locally and navigating to <http://0.0.0.0:8000/openapi.json>, or in the OpenAPI schema [here](/.well-known/openapi.yaml). Note that the OpenAPI schema only contains the `/query` endpoint, because that is the only function that ChatGPT needs to access. This way, ChatGPT can use the plugin on ...
```
I don't think you should make these sort of changes in this PR. keep it relevant to the task

kiwicopple commented on README.md:
```diff
@@ -233,14 +233,14 @@ You can run Weaviate in 4 ways:
+     - `Subscription Tier` – Sandbox for a free trial, or contact [hello@weaviate.io](mailto:hello@weaviate.io) for other options.
+     - `Weaviate Version` - The latest version by default.
+     - `OIDC Authentication` – Enabled by default. This requires a username and password to access your instance.
+  4. Wait for a few minutes until your cluster is ready. You will see a green tick ✔️ when it's done. Copy your cluster URL.
```
Likewise - move this to another PR if you want

kiwicopple commented on README.md:
```diff
@@ -491,6 +491,30 @@ Environment Variables:
+| `BEARER_TOKEN`   | Yes      | Secret token                                 |            |
+| `OPENAI_API_KEY` | Yes      | OpenAI API key                               |            |
+| `SUPABASE_URL`   | Yes      | Supabase Project URL                         |            |
+| `SUPABASE_KEY`   | Yes      | Supabase Project API key (anon or service)   |            |
```
should we split this into 2? `SUPABASE_ANON_KEY` and `SUPABASE_SERVICE_ROLE_KEY`. 

we might need both in the future

logankilpatrick committed:
Merge pull request #41 from eltociear/patch-1

Fix typo in milvus_datastore.py

logankilpatrick committed:
Merge pull request #18 from bojieli/main

small typo fixes to code comments and README

Nasafato committed:
Fix README.md path to qdrant integration tests

pcheng17 committed:
Add whitespace between text from different pages

jamescalam committed:
added additional info for pinecone datastore

jamescalam committed:
added mention for hybrid inside pinecone section

js-ts committed:
Spell Fixes

logankilpatrick committed:
Merge pull request #73 from js-ts/main

Spell Fixes

logankilpatrick committed:
Merge pull request #64 from Nasafato/patch-1

Fix README.md path to qdrant integration tests

logankilpatrick committed:
Merge pull request #31 from po3rin/fix-test-command-path-readme

fix test command in README.md

kacperlukawski committed:
Increase the default timeout in QdrantClient

isafulf committed:
Merge pull request #71 from jamescalam/pinecone-readme-update

added additional info for pinecone datastore

isafulf committed:
Merge pull request #38 from RedisVentures/redis-module-check

add redis module check

isafulf committed:
Merge pull request #76 from kacperlukawski/fix/qdrant-timeout

Increase the default timeout in QdrantClient

isafulf committed:
Setting source to file in upsert file endpoint

Co-Authored-By: Sudharsan <16713484+p2c2e@users.noreply.github.com>
Co-Authored-By: CyberAmyntas <1655459+CyberAmyntas@users.noreply.github.com>

isafulf committed:
correct spelling of Weaviate scopes env variable

Co-Authored-By: João C. Monteiro <5129986+jcmonteiro@users.noreply.github.com>

isafulf committed:
Merge pull request #70 from pcheng17/fix-whitespace

Add whitespace between text from different pages

isafulf committed:
Merge pull request #23 from yunfengsay/patch-3

Remove unused code

isafulf committed:
Merge pull request #14 from js-ts/patch-2

Add docker compose install link and fix the command

isafulf committed:
housekeeping

Co-Authored-By: Glib <71976818+GLEF1X@users.noreply.github.com>

Commit:
fixed minio

Signed-off-by: Filip Haltmayer <filip.haltmayer@zilliz.com>

e4c6 committed:
Update ai-plugin.json

GPT4: In the `ai-plugin.json` file, the `legal_info_url` field should contain a URL, but it currently contains an email address. Update the field with a proper URL or remove it if it's not necessary.

```json
{
  ...
  "legal_info_url": "http://example.com/legal-info"
}

isafulf committed:
Merge pull request #108 from filip-halt/milvus-compose

Milvus docker-compose

isafulf committed:
Merge pull request #110 from e4c6/main

Update ai-plugin.json

tylerhutcherson committed:
Redis improvements and cleanup (#116)

* check module versions and use scan iter

* remove async semaphore usage in favor of pipeline

* readability for module check. transaction=False

* revert embedding response addition

---------

Co-authored-by: DvirDukhan <dvir@redis.com>

isafulf committed:
address https://github.com/openai/chatgpt-retrieval-plugin/issues/109

isafulf committed:
Update poetry.lock

isafulf committed:
upsert-file endpoint accepts metadata (#123)

isafulf committed:
Move vector db setup instructions to docs folder (#124)

* upsert-file endpoint accepts metadata

* update quickstart

* Move vector db setup instructions to docs folder

isafulf committed:
Update README.md

isafulf committed:
move deployment instructions to docs/deployment folder

yhmo committed:
Refine Milvus datastore (#87)

Signed-off-by: groot <yihua.mo@zilliz.com>

Disiok committed:
Add LlamaIndex as a Datastore (#99)

* add dep

* wip

* wip

* wuip

* wip

* update dependency

* wip

* wip

* wip

* wip

* lock

* wip

* wip

* wip

* wup

* wip

* update llama-index

* wip

* refactor docs

* update docs and misc

* update remove unused dep

* wip

* wip

isafulf committed:
Localhost plugin support

Commit:
first redis test (#122)

* first redis test

* fixed test

* pr comment

* added docs

egor-romanov committed:
fix conflicts: move supabase docs from readme

egor-romanov committed:
add supabase to unused deps doc

egor-romanov committed:
add supabase local setup docs and tests

egor-romanov committed:
small improvements in setup.md
